052-cruise-control-api-users.md

```diff
@@ -0,0 +1,204 @@
+# Add ability to create Cruise Control REST API users
+
+This proposal is about giving developers the ability to create REST API users for the Cruise Control REST API.
+This would allow roles and permissions to be defined to allow developers and third-party applications to access the Cruise Control REST API without having to disable HTTP basic authentication.
+
+## Current situation
+
+At this time, Strimzi creates two Cruise Control API users:
+
+* `admin`      - Used by Strimzi for Cruise Control admin operations.
+* `heathcheck` - Used by Strimzi for checking the readiness of the Cruise Control application.
+
+The access to these API users are limited to Strimzi.
+So when HTTP basic authentication is enabled for Cruise Control, no one can access the the Cruise Control API other than Strimzi.
+
+## Motivation
+
+There are a couple use cases where developers would want to access the Cruise Control API without disabling API security:
+
+* Debugging Cruise Control in a production environment
```

Comment:
What does this mean? Debugging in production sounds like a bad idea in the first place. So why should it be supported?

Reply:
> For example, when the issue is environmental, and can't be reproduces in lower environments.

I know it is a weak justification but this was what I was getting at.

> The thing I don't see is which debugging data would be enabled, that are currently not available in other forms (status, logs, events), and how you can use them?

Some of the statistical information surrounding cluster and partition load and user tasks is either only available or more easily gathered through API GET requests.  I have updated the the proposal to address this feedback, let me know what you think!